src/Draco.Compiler/Internal/Symbols/Metadata/MetadataPropertyAccessorSymbol.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Immutable;
+using System.Reflection.Metadata;
+using Draco.Compiler.Internal.Symbols.Generic;
+
+namespace Draco.Compiler.Internal.Symbols.Metadata;
+
+/// <summary>
+/// Represents an property accessor symbol loaded from metadata.
+/// </summary>
+internal sealed class MetadataPropertyAccessorSymbol : MetadataMethodSymbol, IPropertyAccessorSymbol
+{
+    public PropertySymbol Property { get; }
+
+    public MetadataPropertyAccessorSymbol(Symbol containingSymbol, MethodDefinition definition, PropertySymbol property) : base(containingSymbol, definition)
+    {
+        this.Property = property;
+    }
+
+    public override FunctionSymbol GenericInstantiate(Symbol? containingSymbol, ImmutableArray<TypeSymbol> arguments) =>
+        base.GenericInstantiate(containingSymbol, arguments);
+    public override FunctionSymbol GenericInstantiate(Symbol? containingSymbol, GenericContext context) =>
+        new PropertyAccessorInstanceSymbol(containingSymbol, this, context, this.Property.GenericInstantiate(containingSymbol, context));
```

Comment:
Let's fix the duplication here. Do we directly instantiate the accessor? Couldn't we directly call instantiate the property instead and then call to its getter/setter?

The root of the problem is, if you instantiate the prop here, then the parent will be another instance that will reinstantiate the getter and setter, duping all generic props and their accessors